app/middleware/resolve_service.js

```diff
@@ -17,5 +17,15 @@ module.exports = function (req, res, next) {
     return renderErrorView(req, res, 'You do not have the rights to access this service.')
   }
 
+  req.service.hasDirectDebitGatewayAccount = false
```

Comment:
If you need to these kind of things (set a variable, iterate over an array, change that variable, break) usually languages nowadays have one-liners to do all of that. In this case: 
```  req.service.hasDirectDebitGatewayAccount = req.service.gateway_account_ids.some((gatewayAccountId) => gatewayAccountId.startsWith('DIRECT_DEBIT:'))```

Reply:
Great optimisation!